pkg/virt-launcher/virtwrap/network/common.go

```diff
@@ -112,18 +114,28 @@ func (h *NetworkUtilsHandler) SetRandomMac(iface string) (net.HardwareAddr, erro
 		return nil, err
 	}
 
-	changed, err := lmf.SpoofMacSameVendor(iface, false)
-	if err != nil {
-		log.Log.Reason(err).Errorf("failed to spoof MAC for an interface: %s", iface)
-		return nil, err
-	}
+	changed := false
 
-	if changed {
-		mac, err = Handler.GetMacDetails(iface)
+	for i := 0; i < randomMacGenerationAttempts; i++ {
+		changed, err = lmf.SpoofMacSameVendor(iface, false)
```

Comment:
I can't remember.. does SpoofMacSameVendor guarantees that we are getting an address that doesn't equal to the old one?

Reply:
My reading of it suggests that no, we don't. Hence the PR.

Reply:
Oh wait you are asking *if* it returns "changed" that it's indeed changed? Yes, if changed is true, then we have a guarantee that the current MAC address is different from "persistent" MAC address which I think is enough.